src/runtime/lpython_builtin.py

```diff
@@ -764,6 +764,14 @@ def _lpython_str_upper(x: str) -> str:
     return res
 
 
+def _lpython_str_join(s:str, lis:list[str]) -> str:
+    res:str = ""
+    i:i32
+    for i in range(len(lis)):
```

Comment:
Can be directly iterated on the list itself, this way it can be worked on iterables easily. Every iterable can be called using `next` so I think that would be more better.

something like this can be more cleaner I think.

```
for ele in range(list_):
    res+=ele
```

Reply:
If it works, we can use the simpler syntax. LPython doesn't yet work for user defined iterables, but eventually it will I think.